packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -27,67 +33,161 @@ import {
-    const input = container.querySelector('input, select, textarea');
+  static focusFirstInputInContainer(container: HTMLElement | null): void {
+    const input = container?.querySelector('input, select, textarea');
     if (input) {
-      input.focus();
+      (input as HTMLElement).focus();
     }
   }
 
-  static isSameDecimalOptions(options1, options2) {
+  static isSameOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
     return options1?.defaultFormatString === options2?.defaultFormatString;
   }
 
-  static isSameIntegerOptions(options1, options2) {
-    return options1?.defaultFormatString === options2?.defaultFormatString;
+  static isSameDecimalOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
+    return FormattingSectionContent.isSameOptions(options1, options2);
+  }
+
+  static isSameIntegerOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
+    return FormattingSectionContent.isSameOptions(options1, options2);
   }
 
-  static isValidColumnName(name) {
-    return name && DbNameValidator.isValidColumnName(name);
+  static isValidColumnName(name: string): boolean {
+    return (
+      name != null && name !== '' && DbNameValidator.isValidColumnName(name)
+    );
   }
 
-  static isValidFormat(columnType, format) {
+  static isValidFormat(
+    columnType: string,
+    format: Partial<TableColumnFormat>
+  ): boolean {
     // Undefined or empty string formats are always invalid
     if (!columnType || !format.formatString) {
       return false;
     }
 
+    const { formatString } = format;
+    assertNotNull(formatString);
+
+    const nullCheckedFormat = { formatString };
     switch (columnType) {
       case 'datetime':
-        return DateTimeColumnFormatter.isValid(format);
+        return DateTimeColumnFormatter.isValid(nullCheckedFormat);
       case 'decimal':
-        return DecimalColumnFormatter.isValid(format);
+        return DecimalColumnFormatter.isValid(nullCheckedFormat);
       case 'int':
-        return IntegerColumnFormatter.isValid(format);
+        return IntegerColumnFormatter.isValid(nullCheckedFormat);
       default: {
         log.warn('Trying to validate format of unknown type');
         return true;
       }
     }
   }
 
-  static removeFormatRuleExtraProps(item) {
+  static removeFormatRuleExtraProps(
+    item: FormatterItem
+  ): Pick<FormatterItem, 'columnName' | 'columnType' | 'format'> {
     const { id, isNewRule, ...rest } = item;
     return rest;
   }
 
-  static isFormatRuleValidForSave(rule) {
+  static isFormatRuleValidForSave(rule: FormatterItem): boolean {
     return (
-      FormattingSectionContent.isValidColumnName(rule.columnName) &&
-      FormattingSectionContent.isValidFormat(rule.columnType, rule.format)
+      FormattingSectionContent.isValidFormat(rule.columnType, rule.format) &&
+      FormattingSectionContent.isValidColumnName(rule.columnName)
```

Comment:
Not sure why you reversed the order of these calls - I'd keep the order the same, since there's no reason to check if the format is valid if the column name isn't valid